specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json

```diff
@@ -1061,6 +1062,8 @@
             "in": "body",
             "required": true,
             "schema": {
```

Comment:
Still not happy with this part.
This generates a class with a List as its only property -
```
    public partial class SkillNames
    {
        /// <summary> Initializes a new instance of SkillNames. </summary>
        public SkillNames()
        {
            SkillNamesValue = new ChangeTrackingList<string>();
        }

        /// <summary> the names of skills to be reset. </summary>
        public IList<string> SkillNamesValue { get; }
    }
```
Looking for ways to have this parameter as a list itself. Can we do that without breaking the service protocol?

Reply:
I agree it doesn't seem to warrant its own type. @shuyangmsft ?

Reply:
@Mohit-Chakraborty If you want to add `skillNames` as a parameter directly. You can add skillNames parameter like-
```
{
      "name": "skillNames",
       "in": "body",
       "required": true,
       "schema": {
          "type": "array",
           "items": {
           "type": "string"
             }
        },
       "description": "The names of skills to reset."
}
```

Reply:
Thanks @ShivangiReja. Do you know if this will be a breaking change for the service?

Reply:
<html>
<body>
<!--StartFragment-->

❌ 1033 - RemovedProperty | The new version is missing a property found in the old version. Was 'skillNames' renamed or removed?Old: Azure.Search/preview/2021-04-30-Preview/searchservice.json#L1064:15
-- | --

Validation tells me this is a breaking change.
<!--EndFragment-->
</body>
</html>

Reply:
So, we should probably make the change in the client to make things look better.

Reply:
It looks like a breaking change, since previously user was passing the object type in the method and now with this change user will have to pass the list.

Reply:
I reverted the change.

Reply:
I still have an error -
<html>
<body>
<!--StartFragment-->

❌ INVALID_TYPE | Expected type object but found type arrayUrl: Azure.Search/preview/2021-04-30-Preview/searchservice.json#L1064
-- | --


<!--EndFragment-->
</body>
</html>

Reply:
Finally managed to fix all validation errors. The above error was due to a discrepancy between the spec and the example for the ResetSkills operation. I updated the example to follow the design of the spec.